kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/message/MessageEventManager.kt

```diff
@@ -0,0 +1,267 @@
+package com.kotlindiscord.kord.extensions.message
+
+import com.kotlindiscord.kord.extensions.utils.events
+import com.kotlindiscord.kord.extensions.utils.waitFor
+import dev.kord.core.Kord
+import dev.kord.core.behavior.MessageBehavior
+import dev.kord.core.entity.Message
+import dev.kord.core.entity.ReactionEmoji
+import dev.kord.core.event.Event
+import dev.kord.core.event.message.*
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+
+/**
+ * Manager of listeners about the events for a message.
+ * @property message Message responsible of events.
+ * @property timeout Time to stop listening to events after the last received event corresponding to the message.
+ */
+public open class MessageEventManager(
+    public val message: MessageBehavior,
+    private val timeout: Long? = DEFAULT_TIME_LISTENING
+) {
+
+    public companion object {
+        /**
+         * Default time before stop the listening of events.
+         */
+        public const val DEFAULT_TIME_LISTENING: Long = 1000L * 60L * 5L
```

Comment:
I put this variable in the class since it is specific to this class
But if this is the convention for the project, I respect it and move the variable